### PR TITLE
Refactor piped metrics exporter

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

Introducing k8s cloudprovider metrics and k8s livestatestore metrics package

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/cloudprovider/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -101,10 +101,6 @@ type provider struct {
 	initErr          error
 }
 
-func init() {
-	registerMetrics()
-}
-
```
Comment (nghialv) [inline pkg/app/piped/cloudprovider/kubernetes/kubernetes.go:107]: Nice.

**Comment by nghialv** [comment]
Nice. Thank you.
/lgtm

**Comment by nakabonne** [comment]
nits: how about:
- `pkg/app/piped/livestatestore/kubernetes/kubernetesmetrics` -> `pkg/app/piped/livestatestore/kubernetes/metrics`
- `pkg/app/piped/cloudprovider/kubernetes/kubernetesmetrics` -> `pkg/app/piped/cloudprovider/kubernetes/metrics`

I feel a bit redundant

**Comment by khanhtc1202** [comment]
> nits: how about:
> 
> * `pkg/app/piped/livestatestore/kubernetes/kubernetesmetrics` -> `pkg/app/piped/livestatestore/kubernetes/metrics`
> * `pkg/app/piped/cloudprovider/kubernetes/kubernetesmetrics` -> `pkg/app/piped/cloudprovider/kubernetes/metrics`
> 
> I feel a bit redundant

I get your point, just felt the same before 😅 But since we have `cachemetrics` already, I think it's better to have a kind of convention: `xxxmetrics` package means `metrics for xxx (parent package) will be implemented here`. How do you think? 🤔

**Comment by nghialv** [comment]
I prefer to keep that convention. The metrics package name will be `target-package-name` + `metrics`.
This is a special case when 2 `target-package-name` have the same name. (`pkg/app/piped/cloudprovider/kubernetes`, `pkg/app/piped/livestatestore/kubernetes`), but in most cases, this convention helps avoid the need for aliases while importing.

This convention is the same for our test, mock package, `redistest`, `redismock`...
and standard go packages are using the same convention for `httptest`, `httputil` inside `http` package.

**Comment by nakabonne** [comment]
makes sense. no objection to that 👍 
/approve

**Pull request merged**